### ISSUE TITLE
[v9] Uncaught (in promise) TypeError: Cannot read property 'asyncId' of undefined

### ISSUE THREAD

**Issue by brotzky** [react-spring/react-spring#955]
Whenever my `useTransition` hook uses `async await next` it throws an error of 
```
Uncaught (in promise) TypeError: Cannot read property 'asyncId' of undefined
    at _callee6$ (runAsync.ts:81)
...
```

Using Yarn Workspaces with react-spring version:
`"react-spring": "9.0.0-canary.809.5.f01ecc2",`

---
This is my code, based off https://github.com/react-spring/react-spring-examples/blob/pull-750/demos/hooks/notification-hub/index.js

```jsx
const [refMap] = useState(() => new WeakMap())
const config = { tension: 125, friction: 20, precision: 0.1 }

const transitions = useTransition(blocks, {
    initial: null,
    from: { height: 0, opacity: 0, overflow: 'hidden' },
    enter: item => async next => {
      await next({ height: refMap.get(item).offsetHeight, opacity: 1, overflow: 'hidden', config })
    },
    leave: { height: 0, opacity: 0, overflow: 'hidden', config },
    key: block => block.id,
})

 return (
   <div>
      {transitions((style, block) => {
        return (
          <animated.div style={style}>
              <div ref={ref => ref && refMap.set(block, ref)>
                  <Block block={block} />
              </div>
          </animated.div>
        )
     })}
  </div>
)
```

Changing the transition to 
```js
const transitions = useTransition(blocks, {
    initial: null,
    from: { height: 0, opacity: 0, overflow: 'hidden' },
    enter: { height: 100, opacity: 1, overflow: 'hidden', config },
    leave: { height: 0, opacity: 0, overflow: 'hidden', config },
    key: block => block.id,
})
```
resolves the issue but I don't get access to the ref/dynamic height of the element within the transition.

Any ideas if this is a react spring issue or somewhere else? The transition worked in v8. I upgraded because I wanted to be able properly get access to the updated state within item (https://github.com/react-spring/react-spring/issues/949).

could be related to: https://github.com/react-spring/react-spring/issues/920

**Comment by aleclarson** [comment]
You can find the latest canary here:
https://www.npmjs.com/package/react-spring?activeTab=versions

`9.0.0-canary.808.17.55c5691` at the time of writing.

That *should* fix your issue. LMK if it doesn't. 👍

**Comment by brotzky** [comment]
I did try that one, and also tried beta-34 :\

**Comment by brotzky** [comment]
Same issue @aleclarson:

`"react-spring": "9.0.0-canary.808.17.55c5691",`

<img width="715" alt="Screen Shot 2020-04-08 at 7 09 35 PM" src="https://user-images.githubusercontent.com/8007686/78850641-8b904a80-79cc-11ea-9c5e-83b92061bc18.png">

**Comment by aleclarson** [comment]
Lovely. I can get this fixed before the next canary. Do you have a minimal reproducible build?

**Comment by brotzky** [comment]
❤️ Let me try to put one together for you. Thanks for all the help.

**Comment by brotzky** [comment]
@aleclarson here's a repro: https://github.com/brotzky/react-spring-repro-issue-955
let me know if you're able to reproduce it or have questions.

**Comment by brotzky** [comment]
Btw sorry I couldn't use a codesandbox for this :\. I think it could be tied to yarn workspaces or something as it worked with CRA.

**Comment by brotzky** [comment]
@aleclarson is there a rough timeframe for the next canary release? No rush intended, but want to know how to plan the next few weeks for myself. Thanks so much 🙏

**Comment by aleclarson** [comment]
Just tested your repro on the latest code. Should be fixed in `9.0.0-canary.808.18` which I will try to release in a day or two. 👍

**Comment by brotzky** [comment]
Wow, amazing!! thanks so much @aleclarson. No rush -- take your time.

**Comment by aleclarson** [comment]
Now available in v9.0.0-rc.2 #985
